### PR TITLE
[docs-beta] Add workspace.yaml [DOC-437]

### PR DESCRIPTION
**Pull request by PedramNavid**
## Summary & Motivation

Adds a reference doc for workspace.yaml.

I used the following as references:

https://docs.dagster.io/concepts/code-locations/workspace-files
https://github.com/dagster-io/cloud-examples/blob/main/multi-location-project/workspace.yaml 
https://github.com/dagster-io/dagster/blob/618731246cf220c0257f5d030e8c4eaf68b5d8b1/docs/content/concepts/code-locations.mdx#L121 

## How I Tested These Changes

I read them

## Changelog

NOCHANGELOG

### CODE REVIEW

**Comment by PedramNavid** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24881?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#24881** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24881?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#24867** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24867?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24866** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24866?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24863** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24863?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @PedramNavid and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/docs-beta/docs/reference/workspace-yaml.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
+# Workspace File Reference
+
+:::info
+    This reference is only applicable to Dagster OSS, for Dagster Cloud see [the Dagster Cloud Code Locations guide](/dagster-plus/deployment/code-locations)
+:::
+
+The `workspace.yaml` file is used to configure code locations in Dagster. It tells Dagster where to find your code and how to load it.
```
Comment (cmpadden) [inline docs/docs-beta/docs/reference/workspace-yaml.md:7]: Should we call out where this file is stored? (e.g. `$DAGSTER_HOME`)?

File: docs/docs-beta/docs/reference/workspace-yaml.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+**Options:**
+- `module_name`: Name of the Python module to load.
+- Other options are the same as `python_file`.
+
+**Example:**
+~~~yaml
+load_from:
+  - python_module:
+      module_name: hello_world_module.definitions
```
Comment (cmpadden) [inline docs/docs-beta/docs/reference/workspace-yaml.md:60]: Should `python_module` come first if that is the most common approach?

Also, should we call out which approach is most common, or why you would choose one over the other?

**Review by cmpadden — CHANGES_REQUESTED** [review]
Overall looks good--just a couple comments on:
- Where the `workspace.yaml` is saved
- Recommendations on when to choose which approach

File: docs/docs-beta/docs/reference/workspace-yaml.md (hunk 1 above)
Comment (PedramNavid) [reply docs/docs-beta/docs/reference/workspace-yaml.md:7]: good idea

File: docs/docs-beta/docs/reference/workspace-yaml.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    This reference is only applicable to Dagster OSS, for Dagster Cloud see [the Dagster Cloud Code Locations guide](/dagster-plus/deployment/code-locations)
+:::
+
+The `workspace.yaml` file is used to configure code locations in Dagster. It tells Dagster where to find your code and how to load it.
+
+## File Structure
+
+The `workspace.yaml` file uses the following structure:
+
+~~~yaml
```
[the remark this answers is not in the document]
Comment (PedramNavid) [reply docs/docs-beta/docs/reference/workspace-yaml.md:13]: Oh -- this is a leftover from Cursor (it doesn't render ```python blocks inside ```markdown blocks so i told it to use ~~~). 

Will replace!

File: docs/docs-beta/docs/reference/workspace-yaml.md (hunk 2 above)
Comment (PedramNavid) [reply docs/docs-beta/docs/reference/workspace-yaml.md:60]: It's a good question. I am hesitant to include that in a reference, but perhaps it's worth linking to a guide which walks through the most common approach?

**Comment by PedramNavid** [comment]
### Merge activity

* **Oct 1, 8:32 PM EDT**: @PedramNavid started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/24881).

**Pull request merged**